src/librustc_const_eval/diagnostics.rs

```diff
@@ -28,7 +28,7 @@ For example, the following `match` block has too many arms:
 ```compile_fail,E0001
 match Some(0) {
     Some(bar) => {/* ... */}
-    None => {/* ... */}
+    x => {/* ... */} // This handles the `None` case
```

Comment:
ah nevermind, because this triggers for E0001 and not for `unreachable_patterns`